gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -294,10 +335,30 @@ public synchronized void setActive(boolean active) {
     @Override
     public void run() {
       try {
-        Object nextItem = queue.poll();
+        String nextDagToCancel = cancelQueue.poll();
+        //Poll the cancelQueue for a new Dag to cancel.
+        if (nextDagToCancel != null) {
+          log.info("Cancel flow with DagId {}", nextDagToCancel);
+          if (this.dagToJobs.containsKey(nextDagToCancel)) {
+            List<DagNode<JobExecutionPlan>> dagNodesToCancel = this.dagToJobs.get(nextDagToCancel);
+            log.info("Found {} DagNodes to cancel.", dagNodesToCancel.size());
+            for (DagNode<JobExecutionPlan> dagNodeToCancel : dagNodesToCancel) {
+              Properties props = new Properties();
+              if (dagNodeToCancel.getValue().getAzkabanExecId().isPresent()) {
+                String azkabanExecId = dagNodeToCancel.getValue().getAzkabanExecId().get().getExecId();
+                props.put(ConfigurationKeys.AZKABAN_EXEC_ID, azkabanExecId);
+                log.info("Cancel job with azkaban exec id {}.", azkabanExecId);
+              }
+              DagManagerUtils.getSpecProducer(dagNodeToCancel).deleteSpec(null, props);
```

Comment:
Current implementation of deleteSpec in AzkabanSpecProducer will delete the project, which is not what we want. We want to call cancelFlow() on the underlying AzkabanClient. Will deleteSpec implementation be changed later on?

Reply:
Another suggestion to consider: Should we add a deleteSpec(CompletableFuture future) method to SpecProducer interface? or alternately, provide a separate cancelSpec(CompletableFuture future) method? Either way, seems like the SpecProducer implementation should not be exposed inside the DagManager class.

Reply:
done